src/deroff.rs

```diff
@@ -1299,3 +1331,42 @@ fn test_is_white() {
 //         p = pstats.Stats('fooprof')
 //         p.sort_stats('time').print_stats(100)
 //         #p.sort_stats('calls').print_callers(.5, 'startswith')
+
+struct TranslateTable {
+    // map holds a char=>char replacement map. However, since Rust doesn't really have a
+    // str::replace(char, char) and instead uses str::replace(str, str) we end up having to use
+    // Strings.
+    //
+    // For perf:
+    // Using byte strings would probably be way better since we only need to store a single
+    // character, and even UTF-8 it'd be just [u8; 4] which we can freely convert to &[u8] instead
+    // of a using String (24 bytes each)...@TODO
+    map: HashMap<String, String>,
+}
+
+impl TranslateTable {
+    fn new() -> TranslateTable {
+        TranslateTable {
+            map: HashMap::new(),
+        }
+    }
+    fn maketrans(&mut self, from: &str, to: &str) {
+        if from.len() != to.len() {
+            panic!("from and to must be the same length");
+        }
+        for (f, t) in from.chars().zip(to.chars()) {
+            self.map.insert(f.to_string(), t.to_string());
+        }
+    }
+    fn translate(&self, s: &str) -> String {
+        let mut s = s.to_owned();
+        for (k, v) in self.map.iter() {
+            s = (&*s).replace(k, v);
```

Comment:
Would be good to include tests a couple of this logic to ensure it does. E.g., the following fails:
```rust
#[test]
fn test_translate_table() {
    let mut tbl = TranslateTable::new();
    tbl.maketrans("abc", "bca");
    assert_eq!(tbl.translate("abc"), "bca");
}
```
Not sure if any valid roff files ever have shared chars between the `to` and `from` strings, but wouldn't want to rely on that.

Reply:
Obsoleted by #64